store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -101,6 +108,43 @@ public static CarbonReaderBuilder builder(String tablePath, String tableName) {
     return reader.readSchema();
   }
 
+  /**
+   * Read carbonindex file and return the schema
+   * @param indexFilePath complete path including index file name
+   * @return null, if the index file is not present in the path.
+   *         List<ColumnSchema> from the index file.
+   * @throws IOException
+   */
+  public static List<ColumnSchema> readSchemaInIndexFile(String indexFilePath) throws IOException {
```

Comment:
Can you add test case for non  Transactional Table for this API?

Reply:
This test case is redundant. As new API in this PR, just need path. It doesn't matter whether reader is transactional or not. Hence not added